src/msg/async/rdma/Infiniband.cc

```diff
@@ -39,7 +40,7 @@ Port::Port(CephContext *cct, struct ibv_context* ictxt, uint8_t ipn): ctxt(ictxt
   }
 
   lid = port_attr.lid;
-
+  assert(gid_idx < port_attr.gid_tbl_len);
```

Comment:
i'd suggest use `ceph_assert()` here instead. because `assert()` will be optimized out on any Release build. and we do need to fail no matter running Release or Debug build.

Reply:
Agree. Thanks a lot